test/onyx/log/full_test.clj

```diff
@@ -15,29 +16,33 @@
 
 (def peer-config (assoc (:peer-config config) :onyx/id onyx-id))
 
-(def env (onyx.api/start-env env-config))
+(def messaging 
+  (component/start (aeron/aeron {:opts peer-config})))
 
 (def n-peers 20)
 
-(def v-peers (onyx.api/start-peers n-peers peer-config))
+(facts "peers all join and watch each other"
+       (let [env (onyx.api/start-env env-config)
+             peer-group (onyx.api/start-peer-group peer-config)
+             v-peers (onyx.api/start-peers n-peers peer-group)] 
+         (try 
+           (let [ch (chan n-peers)
+                 replica (loop [replica (extensions/subscribe-to-log (:log env) ch)]
+                           (let [position (<!! ch)
+                                 entry (extensions/read-log-entry (:log env) position)
+                                 new-replica (extensions/apply-log-entry entry replica)]
+                             (if (< (count (:pairs new-replica)) n-peers)
+                               (recur new-replica)
+                               new-replica)))]
+             (fact (:prepared replica) => {})
+             (fact (:accepted replica) => {})
+             (fact (set (keys (:pairs replica)))
+                   => (set (vals (:pairs replica))))
+             (fact (count (:peers replica)) => n-peers))
+
+           (finally
```

Comment:
@MichaelDrogalis I was a bit sick of non-shutdown envs. How's this approach for now?